### PR TITLE
code-fixer: extend for ConsumeBytes

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Ref: https://github.com/google/oss-fuzz-gen/issues/575
Ref: https://github.com/google/oss-fuzz-gen/issues/301

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n d-cov-50 -b bug-fdp-consume-buffers -ns 4 -rd

**Comment by DavidKorczynski** [comment]
https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-09-26-652-d-cov-50-bug-fdp-consume-buffers/index.html

**Comment by DavidKorczynski** [comment]
Fixes look to be working e.g.: https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-09-26-652-d-cov-50-bug-fdp-consume-buffers/benchmark/output-libxml2-htmlcreatefileparserctxt/index.html versus before https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-09-26-649-d-cov-43-bug-fdp-consume-buffers/benchmark/output-libxml2-htmlcreatefileparserctxt/index.html

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n d-cov-51 -b bug-fdp-consume-buffers -ns 8 -rd

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n d-cov-251 -b bug-fdp-consume-buffers -ns 4 -rd

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n d-cov-254 -b bug-fdp-consume-buffers -ns 4 -rd

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n d-cov-255 -b bug-fdp-consume-buffers -ns 4 -rd

**Comment by DavidKorczynski** [comment]
The most recent commit made it worse than before: https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-10-02-652-d-cov-255-bug-fdp-consume-buffers/index.html

**Comment by DavidKorczynski** [comment]
The current one works in cases:
- https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-10-02-652-d-cov-256-bug-fdp-consume-buffers/sample/output-opus-opus_packet_unpad/02.html
- https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-10-02-652-d-cov-256-bug-fdp-consume-buffers/sample/output-opus-opus_packet_unpad/03.html

But is no silver bullet:
- https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-10-02-652-d-cov-256-bug-fdp-consume-buffers/sample/output-opus-opus_decoder_get_nb_samples/02.html

This should be good to go at this point -- my take is that it should have positive impact on the next larger runs and reduce FPs

**Review by AdamKorcz — APPROVED** [review]

File: llm_toolkit/code_fixer.py [hunk 1]
```diff
@@ -656,6 +657,28 @@ def _collect_instruction_extern(benchmark: benchmarklib.Benchmark) -> str:
...
+          'IMPORTANT: the harness source code contains a call to '
+          f'`{buffer_method}`. Whenever this function is used, you MUST validate'
+          ' the size of the vector returned, and make sure that the size of the '
+          f'vector is equal to argument given to `{buffer_method}`. If it is '
+          'not equal, the harness should not proceed.\n')
+      instruction += (
+          f'Furthermore, consider changing {buffer_method} to '
+          '`ConsumeRandomLengthString` for creating `char` buffers or strings. '
+          'In most cases, `ConsumeRandomLengthString` is preferred, and '
+          f'should be used instead of {buffer_method}\n')
```
Comment (DonggeLiu) [inline llm_toolkit/code_fixer.py:677]: Nit: Do we need two instructions if the source code contains both `ConsumeBytes` and `ConsumeData`?
As an alternative to shorten the prompt, we can merge them to
```
IMPORTANT: the harness source code contains a call to `ConsumeBytes ` or `ConsumeBytes`. ...
```
if the code contains either.

**Pull request merged**